easylinks/src/main/java/com/google/step/servlets/ManageServlet.java

```diff
@@ -72,7 +73,35 @@ private static void addLink(HttpServletRequest request, HttpServletResponse resp
 
     ServletHelper.DEFAULT_DATASTORE_SERVICE.put(easyLinkEntity);
     response.sendRedirect("/manage.html");
+  }
+
+  private static void editLink(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    long id = Long.parseLong(request.getParameter("ed-id"));
+    String shortcut = ServletHelper.getParameterWithDefault(request, "ed-shortcut", "");
+    String url = ServletHelper.getParameterWithDefault(request, "ed-url", "");
+    if (shortcut.isEmpty() || shortcut.startsWith(PROVIDED_LINK) || url.isEmpty()) {
+      response.getWriter().println("Invalid Input.");
+      return;
+    }
+
+    deleteLinkWithId(id);
+    Entity easyLinkEntity = new Entity("Link");
+    easyLinkEntity.setProperty("shortcut", shortcut);
+    easyLinkEntity.setProperty("url", url);
+    easyLinkEntity.setProperty("email", ServletHelper.USERSERVICE.getCurrentUser().getEmail());
```

Comment:
Maybe don't need to delete the entity first and then add? If you set the key of easyLinkEntity to be id, I think the old entity can be overwritten.

Reply:
Done.